netbox/dcim/forms/filtersets.py

```diff
@@ -210,7 +210,7 @@ class RackFilterForm(TenancyFilterForm, ContactModelFilterForm, NetBoxModelFilte
     model = Rack
     fieldsets = (
         (None, ('q', 'tag')),
-        ('Location', ('region_id', 'site_id', 'location_id')),
+        ('Location', ('region_id', 'site_id', 'site_group_id', 'location_id')),
```

Comment:
Let's move site group before site to maintain hierarchical order.

```suggestion
        ('Location', ('region_id', 'site_group_id', 'site_id', 'location_id')),
```